kubernetes/src/main/scala/org/apache/spark/scheduler/cluster/kubernetes/KubernetesClusterSchedulerBackend.scala

```diff
@@ -97,94 +97,32 @@ private[spark] class KubernetesClusterSchedulerBackend(
 
   def createExecutorPod(executorNum: Int): String = {
     // create a single k8s executor pod.
-    var annotationMap = Map(
-      "pod.beta.kubernetes.io/init-containers" -> raw"""[
-                {
-                    "name": "client-fetch",
-                    "image": "busybox",
-                    "command": ["wget", "-O", "/work-dir/client.jar", "$clientJarUri"],
-                    "volumeMounts": [
-                        {
-                            "name": "workdir",
-                            "mountPath": "/work-dir"
-                        }
-                    ]
-                },
-                {
-                    "name": "distro-fetch",
-                    "image": "busybox",
-                    "command": ["wget", "-O", "/work-dir/spark.tgz", "$sparkDistUri"],
-                    "volumeMounts": [
-                        {
-                            "name": "workdir",
-                            "mountPath": "/work-dir"
-                        }
-                    ]
-                },
-                {
-                    "name": "setup",
-                    "image": "$sparkDriverImage",
-                    "command": ["./install.sh"],
-                    "volumeMounts": [
-                        {
-                            "name": "workdir",
-                            "mountPath": "/work-dir"
-                        },
-                        {
-                            "name": "opt",
-                            "mountPath": "/opt"
-                        }
-                    ]
-                }
-            ]""")
-
-
     val labelMap = Map("type" -> "spark-executor")
     val podName = s"$sparkExecutorName-$executorNum"
     var pod = new PodBuilder()
       .withNewMetadata()
       .withLabels(labelMap.asJava)
       .withName(podName)
-      .withAnnotations(annotationMap.asJava)
       .endMetadata()
       .withNewSpec()
       .withRestartPolicy("OnFailure")
+
       .addNewContainer().withName("spark-executor").withImage(sparkDriverImage)
-      .withImagePullPolicy("Always")
-      .withCommand("/opt/spark/bin/spark-class")
+      .withImagePullPolicy("IfNotPresent")
+      .withCommand("/opt/executor.sh")
```

Comment:
Maybe a combination of both will be just fine. Perhaps these two scripts could be committed into some directory in spark along with this pr (whoever wants to use k8s as master in spark needs to use the spark version with this pr merged into it anyway). (I've taken a look at the two scripts from @erikerlandson repo and they look simple and generic enough). This could be used by default and people that wish to use theirs could bake them into their images using the same (documented) name. My two cents.